.github/workflows/kiali-release.yml

```diff
@@ -0,0 +1,117 @@
+name: Kiali Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      release_type:
+        description: 'Release Type'     
+        required: true
+        default: 'auto' 
+        type: choice
+        options:
+        - auto
+        - minor
+        - snapshot.0
+        - snpashot.1
+        - edge 
+      backend_repo:
+        description: 'The GitHub repo of the back-end sources, in owner/repo format'
+        required: true
+        default: leandroberetta/kiali 
+        type: string
+      quay_name:
+        description: 'The name of the Quay repository to push the release'        
+        type: string
+        default: quay.io/leandroberetta/kiali
+        required: true 
+
+jobs:
+  initialize:
+    name: Initialize
+    runs-on: ubuntu-latest
+    outputs:
+      target_branch: ${{ steps.target_branch.outputs.target_branch }}
+      release_type: ${{ steps.release_type.outputs.release_type || 'auto' }}      
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      # The initialize job gathers variables for later use in jobs.
+      # We are using this technique rather environment variables because at the moment, they won't work with reusable jobs.
+      # A positive side effect of this is that we can print all variables at start for debugging and troubleshooting.
```

Comment:
this is trivial but... this comment was confusing at first because its before the "id: target_branch" so I thought it was commenting on that. I think this a comment for the `Initialize` job as a whole - so this comment should go above line 29 so it is more clear what it is describing. Then again, I'm still not sure :) When it says `this technique`, what technique does it mean?

Reply:
Thanks, I will move it. 

So, another solution that we tried at first was to have an env block at the jobs level, targetting all jobs, but for the reusable jobs, did not work. 

This workaround/technique is to declare needed variables as outputs of the initialize job, then use it in other blocks as envs or outputs directly.